lms/djangoapps/mobile_api/middleware.py

```diff
@@ -0,0 +1,124 @@
+"""
+Middleware for Mobile APIs
+"""
+from datetime import datetime
+from django.core.cache import cache
+from django.http import HttpResponse
+from pytz import UTC
+from mobile_api.mobile_platform import MobilePlatform
+from mobile_api.models import AppVersionConfig
+from mobile_api.utils import parsed_version
+from openedx.core.lib.mobile_utils import is_request_from_mobile_app
+import request_cache
+
+
+class AppVersionUpgrade(object):
+    """
+    Middleware class to keep track of mobile application version being used
+    """
+    LATEST_VERSION_HEADER = 'EDX-APP-LATEST-VERSION'
+    LAST_SUPPORTED_DATE_HEADER = 'EDX-APP-UPGRADE-DATE'
```

Comment:
Also lets change the name of the header to `EDX-AAP-VERSION-LAST-SUPPORTED-DATE`. That will be much more accurate.